src/nemo-devicelock/host/mcedevicelock.cpp

```diff
@@ -379,4 +379,41 @@ void MceDeviceLockAdaptor::setState(int state)
     }
 }
 
+void MceDeviceLockAdaptor::activateTemporaryLockout()
+{
+    if (m_deviceLock->automaticLocking() == -1) {
+        m_deviceLock->sendErrorReply(QDBusError::AccessDenied, QStringLiteral("Device lock not in use"));
+        return;
+    }
+
+    qint64 temporaryLockTimeout = m_deviceLock->temporaryLockTimeout();
+    if (temporaryLockTimeout <= 0 || temporaryLockTimeout > 24 * 60 * 60) {
+        m_deviceLock->sendErrorReply(QDBusError::AccessDenied, QStringLiteral("Invalid temporaryLockTimeout setting"));
+        return;
+    }
+
+    int maximumAttempts = m_deviceLock->maximumAttempts();
+    if (maximumAttempts <= 0 || maximumAttempts > 16) {
+        m_deviceLock->sendErrorReply(QDBusError::AccessDenied, QStringLiteral("Invalid maximumAttempts setting"));
+        return;
+    }
+
+    int currentAttempts = m_deviceLock->currentAttempts();
+    if (currentAttempts < 0 || currentAttempts > 16) {
+        m_deviceLock->sendErrorReply(QDBusError::AccessDenied, QStringLiteral("Invalid currentAttempts setting"));
+        return;
+    }
+
+    if (currentAttempts == maximumAttempts)
+        return;
+
+    uint callerServicePid(m_deviceLock->connection().interface()->servicePid(m_deviceLock->message().service()).value());
+    QFileInfo info(QString("/proc/%1").arg(callerServicePid));
+    bool callerIsRoot(info.owner() == "root");
+    if (!callerIsRoot)
```

Comment:
Should not this be the first/second one to check? And privileged use does not need to use this?

Reply:
I guess the check order ended up being "debugging friendly." Maybe it should be rearranged.

I'd perhaps keep this root only until there is need for privileged user apps to use it.

But now I got thinking: Locking device does not require special privileges. So, I'm wondering if fallback for "lockout" failing due to permissions / similar should be "lock" instead of "nop?"

Reply:
Moved root check earlier